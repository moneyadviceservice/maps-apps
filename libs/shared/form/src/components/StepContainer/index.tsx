export type { Props as StepContainerProps } from './StepContainer';
export { StepContainer } from './StepContainer';
