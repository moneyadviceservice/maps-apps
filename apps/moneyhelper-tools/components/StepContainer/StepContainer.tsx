import { BackLink } from 'components/BackLink';
import { Container, Button } from '@maps-digital/shared/ui';
import { twMerge } from 'tailwind-merge';
import { useTranslation } from '@maps-digital/shared/hooks';
import { DataPath } from 'types';

export type Props = {
  children: JSX.Element;
  backLink?: string;
  lang?: string | string[];
  data?: string;
  action?: string;
  buttonText?: string;
  dataPath?: DataPath;
  isEmbed?: boolean;
  buttonClassName?: string;
  currentStep?: number;
};

export const StepContainer = ({
  children,
  backLink,
  lang,
  data,
  action,
  buttonText,
  dataPath,
  isEmbed,
  buttonClassName,
  currentStep,
}: Props) => {
  const { z } = useTranslation();

  return (
    <Container>
      <div className="space-y-9 pb-7">
        {backLink && (!isEmbed || currentStep !== 1) && (
          <BackLink href={backLink}>{z({ en: 'Back', cy: 'Yn Ã´l' })}</BackLink>
        )}
        {action ? (
          <form method="POST">
            <input
              type="hidden"
              name="isEmbed"
              value={isEmbed ? 'true' : 'false'}
            />
            <input type="hidden" name="language" value={lang} />
            <input type="hidden" name="savedData" value={data} />
            <input type="hidden" name="dataPath" value={dataPath} />
            {children}
            <Button
              className={twMerge('mt-8', buttonClassName)}
              variant="primary"
              formAction={action}
            >
              {buttonText}
            </Button>
          </form>
        ) : (
          children
        )}
      </div>
    </Container>
  );
};
