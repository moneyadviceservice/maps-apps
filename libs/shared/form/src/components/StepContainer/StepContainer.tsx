import { twMerge } from 'tailwind-merge';

import { BackLink } from '@maps-react/common/components/BackLink';
import { Button } from '@maps-react/common/components/Button';
import { Container } from '@maps-react/core/components/Container';
import { useTranslation } from '@maps-react/hooks/useTranslation';

export type Props = {
  children: JSX.Element;
  backLink?: string;
  lang?: string | string[];
  data?: string;
  action?: string;
  buttonText?: string;
  dataPath?: string;
  isEmbed?: boolean;
  buttonClassName?: string;
  currentStep?: number;
  onSubmit?: (e: React.FormEvent<HTMLFormElement>) => void;
};

export const StepContainer = ({
  children,
  backLink,
  lang,
  data,
  action,
  buttonText,
  dataPath,
  isEmbed,
  buttonClassName,
  currentStep,
  onSubmit,
}: Props) => {
  const { z } = useTranslation();

  return (
    <Container>
      <>
        {backLink && (!isEmbed || currentStep !== 1) && (
          <BackLink href={backLink}>{z({ en: 'Back', cy: 'Yn Ã´l' })}</BackLink>
        )}
        {action ? (
          <form data-testid="form" method="POST" onSubmit={onSubmit}>
            <input
              type="hidden"
              name="isEmbed"
              value={isEmbed ? 'true' : 'false'}
            />
            <input type="hidden" name="language" value={lang} />
            <input type="hidden" name="savedData" value={data} />
            <input type="hidden" name="dataPath" value={dataPath} />
            {children}
            <Button
              className={twMerge('mt-8', buttonClassName)}
              variant="primary"
              formAction={action}
              data-testid={'step-container-submit-button'}
            >
              {buttonText}
            </Button>
          </form>
        ) : (
          children
        )}
      </>
    </Container>
  );
};
