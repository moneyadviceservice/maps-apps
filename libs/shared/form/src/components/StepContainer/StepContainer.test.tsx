import React from 'react';

import { fireEvent, render } from '@testing-library/react';

import { StepContainer } from './StepContainer';

jest.mock('@maps-react/hooks/useTranslation', () => ({
  useTranslation: () => ({ z: () => 'Back' }),
}));

describe('StepContainer component', () => {
  it('renders correctly as form', () => {
    const { container } = render(
      <StepContainer
        backLink="test/back/link"
        isEmbed={false}
        onSubmit={undefined}
      >
        <span>Questions</span>
      </StepContainer>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with children only', () => {
    const formSubmit = jest.fn();
    const { container, getByTestId } = render(
      <StepContainer
        isEmbed={true}
        backLink="test/back/link"
        currentStep={2}
        action={'/api/submit'}
        onSubmit={formSubmit}
      >
        <span>Questions</span>
      </StepContainer>,
    );
    const form = getByTestId('form');
    fireEvent.submit(form);
    expect(formSubmit).toHaveBeenCalled();
    expect(container.firstChild).toMatchSnapshot();
  });
});
