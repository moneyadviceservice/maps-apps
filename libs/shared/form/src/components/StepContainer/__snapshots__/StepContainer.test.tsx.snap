// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepContainer component renders correctly as form 1`] = `
<div
  class="container-auto"
>
  <div
    class="flex items-center text-pink-600 group"
  >
    <svgr-mock
      aria-hidden="true"
      classname="text-pink-600 group-hover:text-pink-800 w-[8px] h-[15px]"
    />
    <a
      class="ml-2 underline tool-nav-prev group-hover:text-pink-800 group-hover:no-underline"
      href="test/back/link"
    >
      Back
    </a>
  </div>
  <span>
    Questions
  </span>
</div>
`;

exports[`StepContainer component renders correctly with children only 1`] = `
<div
  class="container-auto"
>
  <div
    class="flex items-center text-pink-600 group"
  >
    <svgr-mock
      aria-hidden="true"
      classname="text-pink-600 group-hover:text-pink-800 w-[8px] h-[15px]"
    />
    <a
      class="ml-2 underline tool-nav-prev group-hover:text-pink-800 group-hover:no-underline"
      href="test/back/link"
    >
      Back
    </a>
  </div>
  <form
    data-testid="form"
    method="POST"
  >
    <input
      name="isEmbed"
      type="hidden"
      value="true"
    />
    <input
      name="language"
      type="hidden"
      value=""
    />
    <input
      name="savedData"
      type="hidden"
      value=""
    />
    <input
      name="dataPath"
      type="hidden"
      value=""
    />
    <span>
      Questions
    </span>
    <button
      class="border-0 text-white bg-pink-600 hover:bg-pink-800 hover:shadow-none active:bg-pink-900 active:outline-yellow-200 active:text-white t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0 t-primary-button mt-8"
      data-testid="step-container-submit-button"
      formaction="/api/submit"
      type="submit"
    >
      <span />
      <span />
    </button>
  </form>
</div>
`;
